src/Neo/SmartContract/Native/ContractManagement.cs

```diff
@@ -106,18 +106,18 @@ internal override async ContractTask OnPersist(ApplicationEngine engine)
         {
             foreach (NativeContract contract in Contracts)
             {
-                uint[] updates = engine.ProtocolSettings.NativeUpdateHistory[contract.Name];
-                if (updates.Length == 0 || updates[0] != engine.PersistingBlock.Index)
-                    continue;
-                engine.Snapshot.Add(CreateStorageKey(Prefix_Contract).Add(contract.Hash), new StorageItem(new ContractState
+                if (contract.IsInitializeBlock(engine.ProtocolSettings, engine.PersistingBlock.Index))
```

Comment:
How to _update_ contracts with this?

Reply:
That's the next step, this pr is only for merge the logics

Reply:
That is good. It will be a great improvement.